main.py

```diff
@@ -2,26 +2,60 @@
 import models
 import datetime
 
+
 def add_heart_rate(email, heart_rate, time):
-    user = models.User.objects.raw({"_id": "suyash@suyashkumar.com"}).first() # Get the first user where _id=suyash@suyashkumar.com
-    user.heart_rate.append(heart_rate) # Append the heart_rate to the user's list of heart rates
-    user.heart_rate_times.append(time) # append the current time to the user's list of heart rate times
-    user.save() # save the user to the database
+    user = models.User.objects.raw({"_id": email}).first()
+    user.heart_rate.append(heart_rate)
+    user.heart_rate_times.append(time)
+    user.save()
+
+
+def create_user(email, age=None, hr=None):
+    u = models.User(email, age, [], [])
+    if hr:
+        u.heart_rate.append(hr)
+        u.heart_rate_times.append(datetime.datetime.now())
+    u.save()
 
-def create_user():
-    u = models.User("suyash@suyashkumar.com", 24, [], []) # create a new User instance
-    u.heart_rate.append(60) # add initial heart rate
-    u.heart_rate_times.append(datetime.datetime.now()) # add initial heart rate time
-    u.save() # save the user to the database
 
 def print_user(email):
-    user = models.User.objects.raw({"_id": "suyash@suyashkumar.com"}).first() # Get the first user where _id=suyash@suyashkumar.com
+    user = models.User.objects.raw({"_id": email}).first()
     print(user.email)
     print(user.heart_rate)
     print(user.heart_rate_times)
 
+
+def user_dict(email):
+    user = models.User.objects.raw({"_id": email}).first()
+    vals = {
+            "user_email": user.email,
+            "user_age": user.age,
+            "heart_rates": user.heart_rate,
+            "heart_rate_times": user.heart_rate_times
+            }
+    return vals
+
+
+def user_heart_rates(email):
+    if already_user(email):
+        user = models.User.objects.raw({"_id": email}).first()
+    else:
+        return None
+    return user.heart_rate
+
+
+def already_user(email):
+    try:
+        user = models.User.objects.raw({"_id": email}).first()
+    except:
```

Comment:
try to catch a particular Exception here.

Reply:
Do you know what Exception is thrown if there is no user that exists with that id? I am having trouble finding source code online. Or, is there a `.exists()` or `.count()` function that can accompany a filter? I'm just having trouble finding documentation in general

Reply:
Just kidding I found a way with...
`return models.User.objects.raw({"_id": email}).count() > 0`